src/lib/config.coffee

```diff
@@ -123,46 +111,28 @@ module.exports = (dnschain) ->
 
     nconf.file 'global', {file:"/etc/#{appname}/#{appname}.conf", format:props}
 
-    # TODO: this blockchain-specific config stuff should be moved out of
-    #       this file and into the constructors (NMCPeer, BDNSPeer, etc.)
-
-    # namecoin
-    nmc = (new nconf.Provider()).argv().env()
-    
-    # TODO: use the same _.find technique as done below for bdnsConf
-    nmcConf = if process.env.APPDATA?
-        path.join process.env.APPDATA, "Namecoin", "namecoin.conf"
-    else if process.env.HOME?
-        path.join process.env.HOME, ".namecoin", "namecoin.conf"
-
-    nmc.file('user', {file:nmcConf,format:props}) if nmcConf
-
-    # bdns
-    bdns = (new nconf.Provider()).argv().env()
-    bdnsConf = _.find _.filter([
-        [process.env.APPDATA, "KeyID"],
-        [process.env.HOME, ".KeyID"],
-        [process.env.HOME, "Library", "Application Support", "KeyID"]]
-    , (x) -> !!x[0])
-    , (x) -> fs.existsSync path.join x...
-
-    if bdnsConf
-        bdns.file 'user', file: path.join(bdnsConf..., 'config.json')
-    else
-        # TODO: this!
-
     stores =
         dnschain: nconf.defaults defaults
-        nmc: nmc.defaults nmcDefs
-        bdns: bdns.defaults bdnsDefs
 
     config =
         get: (key, store="dnschain") -> stores[store].get key
         set: (key, value, store="dnschain") -> stores[store].set key, value
-        # Namecoin's config is not ours, so we don't pretend it is
-        nmc:
-            get: (key)-> config.get key, 'nmc'
-            set: (key, value)-> config.set key, value, 'nmc'
-        bdns:
-            get: (key)-> config.get key, 'bdns'
-            set: (key, value)-> config.set key, value, 'bdns'
+        add: (name, path) ->
+            if stores[name]?
+                return
+            path = [path] if not Array.isArray(path)
+            path.push(stores.dnschain.get("#{name}:config")) if stores.dnschain.get("#{name}:config")?
+            conf = (new nconf.Provider()).argv().env()
+            confFile = _.find path, (x) -> fs.existsSync x
+            conf.file(confFile) if confFile
+            conf.defaults(stores.dnschain.get("#{name}")) if stores.dnschain.get("#{name}")?
+            stores[name] = conf
+            config[name] = conf
+        hosts: ->
+            _.uniq [
+                "127.0.0.", "10.0.0.", "192.168.", "::1", "fe80::"
+                gConf.get('dns:host'), gConf.get('http:host')
+                gConf.get('dns:externalIP'), gExternalIP()
+                _.map(stores, (c) ->
+                    c.get('host'))...
```

Comment:
Two more problems:

1. Having `_.map(stores...` there will result in an an array within an array, which is not what you want. You want to concat onto the existing array that's being created here.
2. This function is being called on each resolution, and since it's performing some iteration it would be better to simply cache the resulting array instead of computing it each time. Remember, however, that you should calculate this value only after all of the blockchain `config:` methods have been called (since they might add the `c.get('host')` value).